test/leak.cc

```diff
@@ -0,0 +1,181 @@
+
+#include <memory>
+#include <string>
+
+#include "include/proxy-wasm/context.h"
+#include "include/proxy-wasm/wasm.h"
+#include "include/proxy-wasm/wasmtime.h"
+
+#include "wasmtime/include/wasm.h"
+#include "wasmtime/include/wasmtime.h"
+
+#include "test/watdata.h"
+#include "gtest/gtest.h"
+#include <chrono>
+#include <fstream>
+#include <thread>
+
+namespace proxy_wasm {
+namespace {
+
+struct DummyIntegration : public WasmVmIntegration {
+  ~DummyIntegration() override{};
+  WasmVmIntegration *clone() override { return new DummyIntegration{}; }
+  void error(std::string_view message) override {
+    std::cout << "ERROR from integration: " << message << std::endl;
+    error_message_ = message;
+  }
+  bool getNullVmFunction(std::string_view function_name, bool returns_word, int number_of_arguments,
+                         NullPlugin *plugin, void *ptr_to_function_return) override {
+    return false;
+  };
+  std::string error_message_;
+};
+
+void callback(void *raw_context) {}
+Word callback2(void *raw_context, Word val) { return val + 100; }
+
+TEST(Wasmtime, CreateVM) {
+  auto raw = wat_data::HelloGo;
```

Comment:
this is the actual wat compiled by TinyGo SDK with which we can introduce memory leak just like V8 in vm_id lotations